hax/data_extractor.py

```diff
@@ -0,0 +1,145 @@
+"""Extract peak or hit info from processed root file
+"""
+
+import numpy as np
+import hax
+
+def root_to_numpy(base_object, field_name, attributes):
+    """Convert objects stored in base_object.field_name to numpy array
+    Will query attributes for each of the objects in base_object.field_name
+    No, root_numpy does not do this for you, that's for trees...
+    """
+    objects_to_convert = getattr(base_object, field_name)
+    if not len(objects_to_convert):
+        return None
+    return np.array([tuple([getattr(p, pa) for pa in attributes]) for p in objects_to_convert])
+
+def build_cut_string(cut_list,obj):
+    '''
+    Build a string of cuts that can be applied using eval() function.
+    '''
+    if len(cut_list) == 0:
+        return 'True'
+    cut_string = '('
+    for cut in cut_list[:-1]:
+        cut_string += obj + '.' + cut + ') & ('
+    cut_string += obj + '.' + cut_list[-1] + ')'
+    return cut_string
+
+def make_branch_selection(level, event_fields, peak_fields, added_branches):
+    """Make the list of branches that have to be selected.
+    """
+    branch_selection_events = event_fields
+    branch_selection_peaks = ['peaks.' + field for field in peak_fields]
+    
+    # For hits, just select the whole hit branch. 
+    # Unfortunately, specifically setting one variable does not seem to work.
+    branch_selection_hits = ['peaks.hits*'] 
+    if level == 'hit':
+        branch_selection = branch_selection_events + branch_selection_peaks + branch_selection_hits + added_branches
+    if level == 'peak':
+        branch_selection = branch_selection_events + branch_selection_peaks + added_branches
+    # Hack to enable range_50p_ar    
+    branch_selection = [field.replace('peaks.range_50p_area','peaks.range_area_decile*') for field in branch_selection]
+    return branch_selection
+    
+def make_named_array(array,field_names):
+    """Make a named array from a numpy array.
+    """
+    import pandas as pd
+    df = pd.DataFrame(array, columns=field_names)
+    array = df.to_records()
+    return array
+    
+
+class DataExtractor():
+    """This class is meant for extracting properties that are *not* on the event level, such as peak or hit properties.
+    """
+    
+    def __init__(self):
+        # Initialize empty data list
+        self.data = []
+        # Default values for cuts: everything passes
+        event_cut_string = 'True'
```

Comment:
__[flake8]__
* [F841] local variable 'event_cut_string' is assigned to but never used